mlkem/rej_uniform.c

```diff
@@ -12,10 +12,11 @@
  *
  * Arguments:   - int16_t *r:          pointer to output buffer
  *              - unsigned int len:    requested number of 16-bit integers
- *                                     (uniform mod q)
+ *                                     (uniform mod q) Must be <= 1024.
  *              - const uint8_t *buf:  pointer to input buffer
  *                                     (assumed to be uniform random bytes)
  *              - unsigned int buflen: length of input buffer in bytes
+ *                                     Must be <= 4096.
```

Comment:
I don't understand where this limitation is coming from? Is this a CBMC limitation? In that case I would not make it part of the docstring here as this is not a real limitation of the C code. 
You can instead add a not to the `REQUIRES` stmt to clarify why that pre-condition exists.

Reply:
This is not a CBMC limitation -- the `pos + 3` calculation can overflow if you have `len == UINT_MAX-2`. I mentioned that in the code.

I don't like getting the doc and spec get out of sync

Reply:
To be fair, the limitation is for `buflen` only, not `inlen`. 

If you prefer, I can state the spec as strong as possible and give the largest possible bound for `buflen`.

Reply:
Ok, left a comment and removed the bound for `len`

Reply:
Thanks for adjusting this!